tests/tools/test_parmetis_preproc.py

```diff
@@ -0,0 +1,236 @@
+    num_chunks : int
+        no. of files to create for each edge type
+    edge_fmt : str, optional
+        to specify whether this file is csv or parquet
+    edge_fmt_del : str optional
+        delimiter to use in the edges file
+
+    Returns:
+    --------
+    dict :
+        dictionary created which represents the schema used for
+        creating the input dataset
+    """
+    schema = {}
+    schema["num_nodes_per_type"] = [10]
+    schema["edge_type"] = ["n1:e1:n1"]
+    schema["node_type"] = ["n1"]
+
+    edges = {}
+    edges["n1:e1:n1"] = {}
+    edges["n1:e1:n1"]["format"] = {}
+    edges["n1:e1:n1"]["format"]["name"] = edge_fmt
+    edges["n1:e1:n1"]["format"]["delimiter"] = edge_fmt_del
+
+    os.makedirs(edges_dir, exist_ok=True)
+    fmt_meta = {"name": edge_fmt}
+    if edge_fmt == "csv":
+        fmt_meta["delimiter"] = edge_fmt_del
+
+    for idx in range(num_chunks):
+        path = os.path.join(edges_dir, f"test_edges_{idx}")
+        array_parser = array_readwriter.get_array_parser(**fmt_meta)
+        edge_data = (
+            np.array([np.arange(10), np.arange(10)]).reshape(10, 2) + 10 * idx
+        )
+        array_parser.write(path, edge_data)
+
+    edge_files = [path]
+    edges["n1:e1:n1"]["data"] = edge_files
+    schema["edges"] = edges
+
+    return schema
+
+
+@pytest.mark.parametrize("num_chunks, num_parts", [[4, 1], [4, 2], [4, 4]])
+@pytest.mark.parametrize("edges_fmt", ["csv", "parquet"])
+def test_gen_edge_files(num_chunks, num_parts, edges_fmt):
+    """Unit test case for the function
+    tools/distpartitioning/parmetis_preprocess.py::gen_edge_files
+
+    Parameters:
+    -----------
+    num_chunks : int
+        no. of chunks the input graph needs to be partititioned into
+    num_parts : int
+        no. of partitions
+    edges_fmt : string
+        specifying the storage format for the edge files
+    """
+
+    # Create the input dataset
+    with tempfile.TemporaryDirectory() as root_dir:
+
+        # Prepare the state information for firing unit test
+        input_dir = os.path.join(root_dir, "chunked-data")
+        output_dir = os.path.join(root_dir, "preproc_dir")
+
+        # Get the parser object
+        fn_params = namedtuple("fn_params", "input_dir output_dir num_parts")
+        fn_params.input_dir = input_dir
+        fn_params.output_dir = output_dir
+        fn_params.num_parts = num_parts
+
+        # Read the input schema
+        schema_map = _get_test_data(input_dir, num_chunks)
+
+        # Get the global node id offsets for each node type
+        # There is only one node-type in the test graph
+        # which range from 0 thru 9.
+        ntype_gnid_offset = {}
+        ntype_gnid_offset["n1"] = np.array([0, 10 * num_chunks]).reshape(1, 2)
+
+        # Iterate over no. of partitions
+        for rank in range(num_parts):
+
+            # Fire the unit test case and get the results
+            actual_results = gen_edge_files(rank, schema_map, fn_params)
```

Comment:
Where is `actual_results` used?

Reply:
Actual_results was used on the line 205. Basically if you look at how this unit test works, we have a list of file paths stored in actual_results and baseline results store the file paths which are used for the input to the 'gen_edge_files' function. So basically if we assert on the length of these two lists it should address the correctness of the unit test. Eitherways, code is now altered to address the CI review comment and now we iterate over the actual_results.